packages/grid/_modules_/grid/components/menu/GridMenu.tsx

```diff
@@ -42,13 +42,35 @@ export const GridMenu: React.FC<MenuProps> = ({
   position,
   ...other
 }) => {
+  const menuListRef = React.useRef<HTMLUListElement | null>(null);
+  const focusTimeout = React.useRef<any>();
+
+  React.useEffect(() => {
+    focusTimeout.current = setTimeout(() => {
+      if (open && menuListRef.current && target) {
+        const firstItem = menuListRef.current.querySelector('li:first-child');
+        if (firstItem) {
+          (firstItem as HTMLLIElement).focus();
+        } else {
+          menuListRef.current.focus();
+        }
+      }
+    }, 0);
+  }, [target, open, menuListRef]);
+
+  React.useEffect(() => {
+    return () => {
+      clearTimeout(focusTimeout.current);
+    };
+  }, []);
```

Comment:
It shouldn't be required, it adds dept

```suggestion
```

Reply:
Yes it does, try the arrows

Reply:
I have tried it on https://deploy-preview-851--material-ui-x.netlify.app/components/data-grid/.

Reply:
> It shouldn't be required, it adds dept
> 
> (and doesn't seem to solve the keyboard issue)

Well the `autofocus` and `autofocusItems` prop doesn't seem to work, this code solves the focus issue. 
In the docs it works, but when you press the down arrow it also scroll, so we need to fix that in a separate PR

Reply:
Please remove this code and the Fragment. We should stick to the example in the documentation. For instance, if you apply:

```diff
diff --git a/packages/grid/_modules_/grid/components/menu/GridMenu.tsx b/packages/grid/_modules_/grid/components/menu/GridMenu.tsx
index b9f84b3..be87dc2 100644
--- a/packages/grid/_modules_/grid/components/menu/GridMenu.tsx
+++ b/packages/grid/_modules_/grid/components/menu/GridMenu.tsx
@@ -2,6 +2,7 @@ import * as React from 'react';
 import ClickAwayListener from '@material-ui/core/ClickAwayListener';
 import Grow from '@material-ui/core/Grow';
 import MenuList from '@material-ui/core/MenuList';
+import MenuItem from '@material-ui/core/MenuItem';
 import Paper from '@material-ui/core/Paper';
 import Popper, { PopperProps } from '@material-ui/core/Popper';

@@ -49,7 +50,9 @@ export const GridMenu: React.FC<MenuProps> = ({
           <Paper>
             <ClickAwayListener onClickAway={onClickAway}>
               <MenuList autoFocusItem={open} id="menu-list-grow" onKeyDown={onKeyDown}>
-                {children}
+                <MenuItem>Option 1</MenuItem>
+                <MenuItem>Option 2</MenuItem>
+                <MenuItem>Option 3</MenuItem>
               </MenuList>
             </ClickAwayListener>
           </Paper>
```

It's already a great first step:

![keyboard](https://user-images.githubusercontent.com/3165635/104745763-b921d580-574e-11eb-9308-fdae129d8e8b.gif)

Then, it's about adding the missing <kbd>Escape</kbd> logic https://material-ui.com/components/menus/#MenuListComposition.tsx

Reply:
Cleaned. As discussed the keyboard issue will be treated in another issue